mdata/aggmetric.go

```diff
@@ -54,99 +58,15 @@ func NewAggMetric(store Store, key string, chunkSpan, numChunks uint32, ttl uint
 func (a *AggMetric) SyncChunkSaveState(ts uint32) {
 	a.Lock()
 	defer a.Unlock()
-	chunk := a.getChunkByT0(ts)
-	if chunk != nil {
-		if LogLevel < 2 {
-			log.Debug("AM marking chunk %s:%d as saved.", a.Key, chunk.T0)
-		}
-		chunk.Saved = true
-	}
-}
-
-/* Get a chunk by its T0.  It is expected that the caller has acquired a.Lock()*/
-func (a *AggMetric) getChunkByT0(ts uint32) *chunk.Chunk {
-	// we have no chunks.
-	if len(a.Chunks) == 0 {
-		return nil
-	}
-
-	currentT0 := a.Chunks[a.CurrentChunkPos].T0
-
-	if ts == currentT0 {
-		//found our chunk.
-		return a.Chunks[a.CurrentChunkPos]
-	}
-
-	// requested Chunk is not in our dataset.
-	if ts > currentT0 {
-		return nil
-	}
-
-	// requested Chunk is not in our dataset.
-	if len(a.Chunks) == 1 {
-		return nil
-	}
-
-	// calculate the number of chunks ago our requested T0 is,
-	// assuming that chunks are sequential.
-	chunksAgo := int((currentT0 - ts) / a.ChunkSpan)
-
-	numChunks := len(a.Chunks)
-	oldestPos := a.CurrentChunkPos + 1
-	if oldestPos >= numChunks {
-		oldestPos = 0
+	if ts > a.lastSaveFinish {
+		a.lastSaveFinish = ts
 	}
-
-	var guess int
-
-	if chunksAgo >= (numChunks - 1) {
-		// set guess to the oldest chunk.
-		guess = oldestPos
-	} else {
-		guess = a.CurrentChunkPos - chunksAgo
-		if guess < 0 {
-			guess += numChunks
-		}
+	if ts > a.lastSaveStart {
```

Comment:
As far as I can see the only call site of `SyncChunkSaveState()` is in `cassandraStore.processWriteQueue()`. When a chunk is pushed into the write queue via `AggMetric.persist()` the `AggMetric.lastSaveStart` is already updated in `persist`, there also can't be a race because `AggMetric.SyncChunkSaveState()` and `AggMetric.persist()` are both surrounded by locks, so is there really any possible case where this condition is true? It seems like that's not possible unless there is another way than `AggMetric.persist()` to push chunks into the write queue, but I can't see one.